arithmetization/src/test/java/net/consensys/linea/zktracer/precompiles/PrecompileUtils.java

```diff
@@ -0,0 +1,128 @@
+import static org.hyperledger.besu.datatypes.Address.*;
+
+import java.math.BigInteger;
+
+import org.hyperledger.besu.datatypes.Address;
+
+public class PrecompileUtils {
+
+  /**
+   * Computes the precompile cost based on the precompile address, arguments size, and r value in
+   * case of BLAKE2F.
+   *
+   * @param precompileAddress the address of the precompile contract.
+   * @param cds the call data size for SHA256, RIPEMD160, ID, and MODEXP. For other precompile
+   *     contracts this value is ignored.
+   * @param bbs the base byte size for MODEXP. For other precompile contracts, this value is
+   *     ignored.
+   * @param mbs the modulus byte size for MODEXP. For other precompile contracts, this value is
+   *     ignored.
+   * @param exponentLog the logarithm of the exponent for MODEXP. For other precompile contracts,
+   *     this value is ignored.
+   * @param r the number of rounds for BLAKE2F. For other precompile contracts, this value is
+   *     ignored.
+   * @return the computed precompile cost.
+   */
+  public static int getPrecompileCost(
+      Address precompileAddress, int cds, int bbs, int mbs, BigInteger exponentLog, int r) {
+    if (precompileAddress.equals(ECREC)) {
+      return getECRECCost();
+    } else if (precompileAddress.equals(SHA256)) {
+      return getSHA256Cost(cds);
+    } else if (precompileAddress.equals(RIPEMD160)) {
+      return getRIPEMD160Cost(cds);
+    } else if (precompileAddress.equals(ID)) {
+      return getIDCost(cds);
+    } else if (precompileAddress.equals(MODEXP)) {
+      return getMODEXPCost(bbs, mbs, exponentLog);
+    } else if (precompileAddress.equals(ALTBN128_ADD)) {
+      return getECADDCost();
+    } else if (precompileAddress.equals(ALTBN128_MUL)) {
+      return getECMULCost();
+    } else if (precompileAddress.equals(ALTBN128_PAIRING)) {
+      return getECPAIRINGCost(cds);
+    } else if (precompileAddress.equals(BLAKE2B_F_COMPRESSION)) {
+      return getBLAKE2FCost(r);
+    } else {
+      throw new IllegalArgumentException("Unknown precompile address");
+    }
+  }
+
+  private static int getECRECCost() {
+    return GAS_CONST_ECRECOVER;
+  }
+
+  private static int getSHA256Cost(int cds) {
+    final int words = (cds + WORD_SIZE_MO) / WORD_SIZE;
+    return GAS_CONST_SHA2 + words * GAS_CONST_SHA2_WORD;
+  }
+
+  private static int getRIPEMD160Cost(int cds) {
+    final int words = (cds + WORD_SIZE_MO) / WORD_SIZE;
+    return GAS_CONST_RIPEMD + words * GAS_CONST_RIPEMD_WORD;
+  }
+
+  private static int getIDCost(int cds) {
+    final int words = (cds + WORD_SIZE_MO) / WORD_SIZE;
+    return GAS_CONST_IDENTITY + words * GAS_CONST_IDENTITY_WORD;
+  }
+
+  static int getMODEXPCost(int bbs, int mbs, BigInteger exponentLog) {
+    final int words = (Math.max(bbs, mbs) + 7) / 8;
+    final int fOfMax = words * words;
+    final BigInteger bigNumerator =
+        BigInteger.valueOf(fOfMax).multiply(exponentLog.max(BigInteger.ONE));
```

Comment:
Why is it that we use `BigInteger`'s here ? With our limitations we have that
- `fOfMax` is at most `64 * 64 = 4096`
- `exponentLog` is at most `256 + 8 * (512 - 32) = 4096`
- `bigQuotient` is at most `5592405`

Reply:
It seems in any case that an `int` should suffice, and that `long`'s would be plenty.